subsquid/src/dbHelper.ts

```diff
@@ -841,6 +845,87 @@ export async function getHistoricalCoingeckoPrice(
   throw new Error("Failed to fetch historical price");
 }
 
+export async function getAllHistoricalCoingeckoPrices(
+  ctx: Context | EntityManager,
+  assetId: string
+): Promise<Array<[Date, number]>> {
+  const assetInfo = assetList.find(asset => asset.assetId === assetId);
+  if (!assetInfo) {
+    throw new Error(`Asset ${assetId} not found in asset list.`);
+  }
+  let coinId: string;
+  if (assetInfo.coingeckoId) {
+    coinId = assetInfo.coingeckoId;
+  } else if (assetInfo.spotPriceBaseAsset) {
+    coinId = assetInfo.spotPriceBaseAsset.coingeckoId;
+  } else {
+    throw new Error("No Coingecko ID found");
+  }
```

Comment:
L856-863 can be minimized: 

const coinId = assetInfo.coingeckoId || assetInfo.spotPriceBaseAset?.coingeckoId
if (!coinId) {
    throw new Error("No Coingecko ID found");
}

Reply:
True. Will update.